pkg/aws/eni/limits/limits.go

```diff
@@ -867,19 +128,68 @@ func UpdateFromEC2API(ctx context.Context, api ec2API) error {
 	return nil
 }
 
-// parseLimitString returns the Limits struct parsed from config string.
-func parseLimitString(limitString string) (limit ipamTypes.Limits, err error) {
-	intSlice := make([]int, 3)
-	stringSlice := strings.Split(strings.ReplaceAll(limitString, " ", ""), ",")
-	if len(stringSlice) != 3 {
-		return limit, fmt.Errorf("invalid limit value")
+func (l *LimitsGetter) newLimitsUpdateTrigger(api ec2API, minInterval, timeout time.Duration, retryCount int) (*trigger.Trigger, error) {
+	return trigger.NewTrigger(trigger.Parameters{
+		Name:        "ec2-instance-limits-update",
+		MinInterval: minInterval,
+		TriggerFunc: func(reasons []string) {
+			if err := l.updateFromEC2APIWithRetry(context.TODO(), api, timeout, retryCount); err != nil {
+				log.WithError(err).Warning("Failed to update instance limits from EC2 API")
+			}
+
+			l.Lock()
+			l.lastUpdate = time.Now()
+			oldDone := l.triggerDone
+			l.triggerDone = make(chan struct{})
+			l.Unlock()
+			if oldDone != nil {
+				close(oldDone)
+			}
+		},
+	})
+}
+
+func (l *LimitsGetter) initEC2APIUpdateTrigger(api ec2API, triggerMinInterval, ec2APITimeout time.Duration, ec2APIRetryCount int) error {
+	l.Lock()
+	l.triggerMinInterval = triggerMinInterval
+	l.ec2APITimeout = ec2APITimeout
+	l.ec2APIRetryCount = ec2APIRetryCount
+	l.Unlock()
+
+	t, err := l.newLimitsUpdateTrigger(api, triggerMinInterval, ec2APITimeout, ec2APIRetryCount)
+	if err != nil {
+		log.WithError(err).Warning("Failed to create EC2 update trigger")
+		return err
 	}
-	for i, s := range stringSlice {
-		intLimit, err := strconv.Atoi(s)
-		if err != nil {
-			return limit, err
-		}
-		intSlice[i] = intLimit
+
+	l.Lock()
+	l.limitsUpdateTrigger = t
+	l.Unlock()
+
+	if err := l.updateFromEC2APIWithRetry(context.Background(), api, ec2APITimeout, ec2APIRetryCount); err != nil {
+		log.WithError(err).Warning("Failed initial EC2 API limits update")
+		return err
 	}
-	return ipamTypes.Limits{Adapters: intSlice[0], IPv4: intSlice[1], IPv6: intSlice[2]}, nil
+	log.Info("Initial EC2 API limits update completed successfully")
+
+	l.Lock()
+	l.lastUpdate = time.Now()
```

Comment:
nit: Both calls to `updateFromEC2APIWithRetry` do update `l.lastUpdate` as soon as `updateFromEC2APIWithRetry` has returned.

This indicates that could move this `l.lastUpdate` write into `updateFromEC2API`